app/controllers/cart_controller.rb

```diff
@@ -4,24 +4,24 @@ class CartController < BaseController
   before_filter :check_authorization
 
   def populate
+    order = current_order(true)
+
     # Without intervention, the Spree::Adjustment#update_adjustable callback is called many times
     # during cart population, for both taxation and enterprise fees. This operation triggers a
     # costly Spree::Order#update!, which only needs to be run once. We avoid this by disabling
     # callbacks on Spree::Adjustment and then manually invoke Spree::Order#update! on success.
     Spree::Adjustment.without_callbacks do
-      cart_service = CartService.new(current_order(true))
+      cart_service = CartService.new(order)
 
       if cart_service.populate(params.slice(:products, :variants, :quantity), true)
-        fire_event('spree.cart.add')
-        fire_event('spree.order.contents_changed')
```

Comment:
I removed these two event calls here. I can't find anything anywhere in `spree_core` or in our codebase that actually listens on the `spree.cart.add` event, and the only listener we have on `spree.order.contents_changed` just calls `#update_distribution_charge!` on the current order, so I've added that call explicitly (below) in place of the event.

Also we were using `current_order` multiple times here, and calling things on it, and it seemed to be re-fetching the order every time.

Reply:
hey, lets burn all the fire_events! It's worth doing it now that we have verified they are not used anywhere in Spree!
There are only two more:

app/controllers/checkout_controller.rb:
   50:     fire_event('spree.checkout.update')

This :point_up: can just be deleted

app/controllers/spree/orders_controller.rb:
  ```
        if @order == current_order
          fire_event('spree.order.contents_changed')
        else
          @order.update_distribution_charge!
        end
```
Reading the code, I'd just replace these lines with
@order.update_distribution_charge!

Reply:
I think this is out of scope for this PR and also it will require very different testing (checkout, etc) so I created a tech debt PR for this:
#5324